src/services/package-manager/variable-service.ts

```diff
@@ -33,9 +36,21 @@ class VariableService {
         const parsedParams = this.parseParams(params);
         const assignments = await variablesApi.getCandidateAssignments(type, parsedParams);
 
-        const filename = uuidv4() + ".json";
-        fileService.writeToFileWithGivenName(JSON.stringify(assignments), filename);
-        logger.info(FileService.fileDownloadedMessage + filename);
+        this.exportToJson(assignments)
+    }
+
+    public async listVariables(keysByVersion: string[], keysByVersionFile: string): Promise<void> {
+        const variableManifests = await this.getVersionedVariablesByKeyVersionPairs(keysByVersion, keysByVersionFile);
+
+        variableManifests.forEach(variableManifest => {
+            logger.info(JSON.stringify(variableManifest));
+        });
+    }
+
+    public async exportVariables(keysByVersion: string[], keysByVersionFile: string): Promise<void> {
+        const variableManifests = await this.getVersionedVariablesByKeyVersionPairs(keysByVersion, keysByVersionFile);
+
+        this.exportToJson(variableManifests);
```

Comment:
Maybe this is a convention for how we do this, but I think it would be easier to remove the responsibility of printing from these services, and handle that in the command classes, e.g. `ConfigCommand`.
That way we would not need to have two method that are essentially the same, only difference is how they are consumed. And it would be very simple to know which method of the service to call.

What do you think does it make sense to leave the export/logging out of the services, or is the effort to change the existing services just not worth it?

Reply:
Logging is used on almost all services so I don't think it makes sense to work on it for now. I added it as a topic for improvement on this [doc](https://docs.google.com/document/d/1vvqThDSm09_Ej_adwIpKSYSxadBgZJt-jC3nT6I919o).